internal/status.go

```diff
@@ -0,0 +1,158 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/dustin/go-humanize"
+)
+
+type StatusLine struct {
+	resources              *[]Resource
+	resourceSizes          []int64
+	totalBytes             int64
+	numBytesDownloaded     int64
+	numResourcesDownloaded int
+	spinI                  int
+	indexCh                chan int
+	startTime              time.Time
+	sizingErr              error
+	ctx                    context.Context
+	mtx                    sync.Mutex
+}
+
+var spinChars = [5]string{"-", "\\", "|", "/", "-"}
+var timeoutMs = 1000
+
+// NewStatusLine creates and initializes a new StatusLine.
+func NewStatusLine(ctx context.Context, resources *[]Resource) (*StatusLine, error) {
+	st := StatusLine{}
+	st.resources = resources
+	st.indexCh = make(chan int)
+	st.ctx = ctx
+	st.sizingErr = st.initResourcesSizes()
+	return &st, st.sizingErr
+}
+
+// Increment informs the StatusLine that a resource (at index i in resource list) has finished downloading.
+func (st *StatusLine) Increment(i int) {
+	st.indexCh <- i
+}
+
+// Start begins the goroutine and loop that will update/print the status line.
+// Pass true to force SL to update (spinner and second counter) every 50ms.
+func (st *StatusLine) Start(doTick bool) {
+	st.startTime = time.Now()
+	go func() {
+		fmt.Print(st.GetStatusString())
+		for {
+			// Block until value is inserted into indexCh (>= 0 when resource finishes downloading, -1 every 50 milliseconds to keep timer and spinner updating).
+			var i int
+			select {
+			case i = <-st.indexCh:
+			case <-time.After(50 * time.Millisecond):
```

Comment:
I think you want to make this duration a const. This way you can refer to it in the tests if needed.